File: Problems/1318. (M) Minimum Flips to Make a OR b Equal to c/Python3/solution.py
class Solution:
  def minFlips(self, a: int, b: int, c: int) -> int:
    # bit manipulation
    
    ret = 0
    while a or b or c:
      c_a = a & 1
      c_b = b & 1
      c_c = c & 1
      if c_c:
        ret += 1 if not (c_a | c_b) else 0
      else:
        ret += 1 if c_a else 0
        ret += 1 if c_b else 0
      a >>= 1
      b >>= 1
      c >>= 1
    
    return ret

File: Problems/1318. (M) Minimum Flips to Make a OR b Equal to c/Java/solution.java
class Solution {
  public int minFlips(int a, int b, int c) {
    // bit manipulation

    int ret = 0;
    while (a > 0 || b > 0 || c > 0) {
      int c_a = a & 1;
      int c_b = b & 1;
      int c_c = c & 1;
      if (c_c == 1)
        ret += (c_a | c_b) == 0 ? 1 : 0;
      else {
        ret += c_a == 1 ? 1 : 0;
        ret += c_b == 1 ? 1 : 0;
      }
      a >>= 1;
      b >>= 1;
      c >>= 1;
    }
    return ret;
  }
}
